Add new Redfish config module to handle Dell-specific OEM extensions

jose-delarosa opened the pull request:
- Extends the existing RedfishUtils class and leverages its methods

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

Implements code taken out by https://github.com/ansible/ansible/pull/51898 and provides module to implement all iDRAC-specific configuration settings. 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
idrac_redfish_config

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

This new module can be used as a model for other OEMs to implement their own Redfish extensions.

Commit:
Add new Redfish config modules to handle Dell-specific OEM extensions

- Extends the existing RedfishUtils class and leverages its methods

Commit:
Removed extra password key

jamescassell commented:
Is it possible to have it be a vendor-neutral redfish module?

jose-delarosa commented:
@jamescassell There are already vendor-neutral modules in this same directory. This Dell-specific module implements code that was taken out of those vendor-neutral modules, and it follows a framework we discussed to implement OEM-specific Redfish implementations. Refer to https://github.com/ansible/ansible/issues/48364#issuecomment-439129431.

billdodd reviewed (commented):
Since the 'timeout' PR has now been merged, this PR will need to be updated to add the timeout support. See the recent timeout related commit to idrac_redfish_command.py.

jose-delarosa commented:
Yeah I realized that yesterday, will update today, thanks Bill.

Commit:
Add timeout parameter

billdodd reviewed (approved):
Looks good.

shipit

jose-delarosa commented:
shipit

gundalow commented on lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py:
```diff
@@ -0,0 +1,222 @@
+    description:
+      - name of iDRAC attribute to update
+    default: 'null'
+    version_added: "2.8"
```
```suggestion
```

gundalow commented on lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py:
```diff
@@ -0,0 +1,222 @@
+    description:
+      - value of iDRAC attribute to update
+    default: 'null'
+    version_added: "2.8"
```
```suggestion
```

gundalow commented on lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py:
```diff
@@ -0,0 +1,222 @@
+      - Timeout in seconds for URL requests to iDRAC controller
+    default: 10
+    type: int
+    version_added: '2.8'
```
```suggestion
```

gundalow commented on lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py:
```diff
@@ -0,0 +1,222 @@
+DOCUMENTATION = '''
+---
+module: idrac_redfish_config
+version_added: "2.8"
```
```suggestion
version_added: '2.8'
```

gundalow commented on lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py:
```diff
@@ -0,0 +1,222 @@
+options:
+  category:
+    required: true
+    description:
```
```suggestion
    type: str
    description:
```

jose-delarosa replied:
Updates committed. Thank you @gundalow

Commit:
Updates to DOCUMENTATION section

- Removed version_added from several parameters
- Added type to category parameter
